### PR TITLE
Fix iterator invalidation Issue in vector erase operations during histories cleanup

### PR DESCRIPTION
**Pull request by JesusEV**
Problem Description
This PR addresses a potential issue where iterators become invalidated when erase() is called on a std::vector. As per [1], after an erase operation, any iterator or reference at or after the point of the erase in the vector is invalidated.

Proposed Solution
The updated implementation uses the fact that erase() returns a new, valid iterator pointing to the element that follows the last element erased [2]. By using this returned iterator, we can safely continue iterating through the vector without encountering invalidated iterators.

References
[1]  https://en.cppreference.com/w/cpp/container/vector/erase
[2] https://cplusplus.com/reference/vector/vector/erase/

### CODE REVIEW

File: nestkernel/eprop_archiving_node.cpp [hunk 1]
```diff
@@ -162,18 +162,17 @@ nest::EpropArchivingNode::get_firing_rate_reg_history( const long time_step )
...
-    // of synapses that require information from that particular update time
-
     if ( it_hist->access_counter_ == 0 )
     {
-      // If access_counter_ is 0, it implies that no synapses need information from this update time,
-      // therefore the corresponding update history entry is no longer needed and can be removed
-      // from the update history. This ensures update history only contains only
-      // entries that are needed by at least one synapse
-      update_history_.erase( it_hist );
+      // erase() invalidates the iterator, but returns a new, valid iterator
```
Comment (akorgor) [inline nestkernel/eprop_archiving_node.cpp:170]: Did you remove here the comments with the background information on purpose? If yes, maybe we could keep them in a shortened version?

File: nestkernel/eprop_archiving_node.cpp [hunk 2]
```diff
@@ -218,16 +217,17 @@ nest::EpropArchivingNode::erase_unneeded_firing_rate_reg_history()
...
-        ++it_update_hist, ++it_reg_hist )
+  while ( it_update_hist != update_history_.end() and it_reg_hist != firing_rate_reg_history_.end() )
   {
     if ( it_update_hist->access_counter_ == 0 )
     {
-      // If no synapses require updates for this time period (indicated by access_counter_ being 0),
-      // the corresponding entry in the firing rate regularization history is no longer needed
-      // and can be safely remove
-      firing_rate_reg_history_.erase( it_reg_hist );
+      it_reg_hist = firing_rate_reg_history_.erase( it_reg_hist );
```
Comment (akorgor) [inline nestkernel/eprop_archiving_node.cpp:224]: Did you remove here the comments with the background information on purpose? If yes, maybe we could keep them in a shortened version?

**Comment by akorgor** [comment]
Overall this change looks good and important to me! If my two minor comments are resolved, we can merge it from my side.

File: nestkernel/eprop_archiving_node.cpp (hunk 1 above)
Comment (JesusEV) [reply nestkernel/eprop_archiving_node.cpp:170]: Yes, I removed them on purpose. Of course updated comments can be added in a new commit

**Pull request merged**